controllers/upgradeconfig/upgradeconfig_controller.go

```diff
@@ -87,12 +87,32 @@ func (r *ReconcileUpgradeConfig) Reconcile(ctx context.Context, request reconcil
 		return reconcile.Result{}, err
 	}
 
+	// Get current ClusterVersion
+	cvClient := r.CvClientBuilder.New(r.Client)
+	clusterVersion, err := cvClient.GetClusterVersion()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
 	history := instance.Status.History.GetHistory(instance.Spec.Desired.Version)
 	if history == nil {
-		history = &upgradev1alpha1.UpgradeHistory{Version: instance.Spec.Desired.Version, Phase: upgradev1alpha1.UpgradePhaseNew}
+		upgraded, err := cvClient.HasUpgradeCommenced(instance)
```

Comment:
I am confused here. So if the instance has been deleted, how we can still identify whether the cluster is upgraded or not? Refer to the above line 78, it will stuck in the reconciliation with "Reset metrics due to no upgrade config present.". Is it here because we always [refresh](https://github.com/openshift/managed-upgrade-operator/blob/master/pkg/upgradeconfigmanager/upgradeconfigmanager.go#L180) to create/replace the UC? From [here](https://github.com/openshift/managed-upgrade-operator/blob/master/main.go#L299)?

Reply:
Indeed, you got it! So, there is a separate goroutine which runs the "upgradeconfig manager", whose job it is to sync with OCM every `X` minutes and check if there is an upgrade policy in OCM, and if so, transform that into an `UpgradeConfig` on the cluster.

Prior to this change, whenever the cluster was in the middle of an upgrade, the upgradeconfig manager would deliberately stop syncing with OCM.

Now with this change, it will continue to sync with OCM all the way through the upgrade. So, if the UpgradeConfig has been removed, but there's still an upgradepolicy for it in OCM - it will eventually reappear as an UpgradeConfig on the cluster.

Now, in terms of the line you've highlighted - what's happening there is some logic to ensure that the UpgradeConfig gets a `status.history` filled out that correctly represents the status of the upgrade. If there's no history at all, then the UpgradeConfig must be new on the cluster and this is MUO's first time reconciling it. If the cluster is in the middle of or completing an upgrade ("is the UpgradeConfig's version the same as CVO's version?"), then we want to set the `history` phase to `Upgrading`. If the cluster is not yet upgrading, then we set the `history` phase to `New`.